torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -130,6 +130,80 @@ std::vector<std::shared_ptr<SugaredValue>> SimpleValue::asTuple(
                          << " cannot be used as a tuple";
 }
 
+void UserTypeValue::assign(
+    const SourceRange& loc,
+    Method& m,
+    const std::string& field,
+    Value* newValue) {
+  auto expectedType = type_->getAttribute(field);
+  if (!expectedType) {
+    // We don't have an attribute with this name, either add it to the type
+    // definition or throw an error
+    if (m.name() == "__init__") {
+      type_->addAttribute(field, newValue->type());
```

Comment:
probably also want to check that the current insertion point is not in a sub-block, indicating that this would be a conditional assignment.